engine/table/src/main/java/io/deephaven/engine/table/impl/perf/UpdatePerformanceStreamPublisher.java

```diff
@@ -0,0 +1,115 @@
+import io.deephaven.engine.table.impl.AbstractColumnSource;
+import io.deephaven.engine.table.impl.perf.UpdatePerformanceTracker.IntervalLevelDetails;
+import io.deephaven.engine.table.impl.util.EngineMetrics;
+import io.deephaven.stream.StreamConsumer;
+import io.deephaven.stream.StreamPublisher;
+import io.deephaven.stream.StreamToBlinkTableAdapter;
+import io.deephaven.time.DateTimeUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.Objects;
+
+class UpdatePerformanceStreamPublisher implements StreamPublisher {
+    private static final TableDefinition DEFINITION = TableDefinition.of(
+            ColumnDefinition.ofString("ProcessUniqueId"),
+            ColumnDefinition.ofInt("EntryId"),
+            ColumnDefinition.ofInt("EvaluationNumber"),
+            ColumnDefinition.ofInt("OperationNumber"),
+            ColumnDefinition.ofString("EntryDescription"),
+            ColumnDefinition.ofString("EntryCallerLine"),
+            ColumnDefinition.ofTime("IntervalStartTime"),
+            ColumnDefinition.ofTime("IntervalEndTime"),
+            ColumnDefinition.ofLong("IntervalDurationNanos"),
+            ColumnDefinition.ofLong("EntryIntervalUsage"),
+            ColumnDefinition.ofLong("EntryIntervalCpuNanos"),
+            ColumnDefinition.ofLong("EntryIntervalUserCpuNanos"),
+            ColumnDefinition.ofLong("EntryIntervalAdded"),
+            ColumnDefinition.ofLong("EntryIntervalRemoved"),
+            ColumnDefinition.ofLong("EntryIntervalModified"),
+            ColumnDefinition.ofLong("EntryIntervalShifted"),
+            ColumnDefinition.ofLong("EntryIntervalInvocationCount"),
+            ColumnDefinition.ofLong("MinFreeMemory"),
+            ColumnDefinition.ofLong("MaxTotalMemory"),
+            ColumnDefinition.ofLong("Collections"),
+            ColumnDefinition.ofLong("CollectionTimeNanos"),
+            ColumnDefinition.ofLong("EntryIntervalAllocatedBytes"),
+            ColumnDefinition.ofLong("EntryIntervalPoolAllocatedBytes"));
+
+    public static TableDefinition definition() {
+        return DEFINITION;
+    }
+
+    private static final int CHUNK_SIZE = 4096; // todo
+
+    private WritableChunk<Values>[] chunks;
+    private StreamConsumer consumer;
+
+    public UpdatePerformanceStreamPublisher() {
+        chunks = StreamToBlinkTableAdapter.makeChunksForDefinition(DEFINITION, CHUNK_SIZE);
+    }
+
+    @Override
+    public void register(@NotNull StreamConsumer consumer) {
+        if (this.consumer != null) {
+            throw new IllegalStateException("Can not register multiple StreamConsumers.");
+        }
+        this.consumer = Objects.requireNonNull(consumer);
+    }
+
+    public synchronized void add(IntervalLevelDetails intervalLevelDetails, PerformanceEntry performanceEntry)
+            throws IOException {
+        chunks[0].<String>asWritableObjectChunk().add(EngineMetrics.getProcessInfo().getId().value());
```

Comment:
We can probably skip this column for DHC;  I don't think there is much value?

Reply:
Ultimately, this field is in this column due to the nature of importing data into the centralized DIS for DHE - for DHC when we are querying it directly, we should prefer to present the data in its "normalized" form; if we need the process id on the table, we should make that easy for a user to do.